### ISSUE TITLE
Deadlock when releasing video source on Win32

### ISSUE THREAD

**Issue by fiban-havok** [microsoft/winrtc#54]
I am running MR-WebRTC built on top of the m84 version of WinRTC, from the Unity editor (so Win32 platform) on x64. The issue seems to happen both when using Debug and Release.

When calling `StopCapture()` on a `webrtc::VideoCaptureModule` previously created with `webrtc::VideoCaptureFactory::Create(const char* deviceUniqueIdUTF8)`, I get this warning
> (help_functions_winrt.cc:430): Waiting in a non-MTA thread. Deadlocks might occur.

and after that the thread indeed hangs forever and the process freezes.

The callstack is:
```
mrwebrtc.dll!webrtc::videocapturemodule::WaitForAsyncAction(ABI::Windows::Foundation::IAsyncAction * p_async_action=0x000001ec79b79d30) Line 430	C++
mrwebrtc.dll!webrtc::videocapturemodule::VideoCaptureWinRTInternal::StopCapture() Line 407	C++
mrwebrtc.dll!webrtc::videocapturemodule::VideoCaptureWinRT::StopCapture() Line 675	C++
(MR-WebRTC stuff follows)
```
The value of `apt_type` is `APTTYPE_MAINSTA`.

I seem to get this both in Debug and in Release (in Release without the error message).

**Comment by LoadLibrary** [comment]
Investigating

**Comment by LoadLibrary** [comment]
It wasn't a deadlock. The async operation was finalizing before the code for waiting for the signal had chance to run. The app would unblock after 3 minutes. A fix for this issue has been pushed.

**Comment by fiban-havok** [comment]
Note that this is __not__ fixed - stopping the capture from the MAINSTA thread will still freeze until the timeout expires (now 5s), even if the operation has completed successfully in the meanwhile. 

@LoadLibrary please reopen this or justify and document if wontfix.
